### PR TITLE
[binder] Fix crash when missing authority

### PR DESCRIPTION
**Pull request by TaWeiTu**
There is an assertion in `Server::CallData::RecvInitialMetadataReady()`
that checks whether the received initial metadata indeed contain keys
`:authority` and `:path`. However, there was no such check in the binder
transport. This PR calls `recv_initial_metadata_ready` with an error when
either `:authority` or `:path` is missing in the received initial metadata.

### CODE REVIEW

File: src/core/ext/transport/binder/transport/binder_transport.cc [hunk 1]
```diff
@@ -172,6 +172,20 @@ static void cancel_stream_locked(grpc_binder_transport* gbt,
   GRPC_ERROR_UNREF(error);
 }
 
+static bool ContainsAuthorityAndPath(const grpc_binder::Metadata& metadata) {
```
Comment (TaWeiTu) [inline src/core/ext/transport/binder/transport/binder_transport.cc:175]: There are the relevant changes of this PR. @ctiller can you confirm that this check is indeed required and correct?

File: src/core/ext/transport/binder/transport/binder_transport.cc [hunk 2]
```diff
@@ -172,6 +172,20 @@ static void cancel_stream_locked(grpc_binder_transport* gbt,
...
+    if (kv.first == grpc_core::StringViewFromSlice(GRPC_MDSTR_AUTHORITY)) {
+      has_authority = true;
+    }
+    if (kv.first == grpc_core::StringViewFromSlice(GRPC_MDSTR_PATH)) {
+      has_path = true;
+    }
+  }
+  return has_authority && has_path;
+}
+
```
Comment (TaWeiTu) [inline src/core/ext/transport/binder/transport/binder_transport.cc:188]: There are the relevant changes of this PR. @ctiller can you confirm that this check is indeed required and correct?

Comment (ctiller) [reply src/core/ext/transport/binder/transport/binder_transport.cc:188]: Path is certainly required... although Authority feels much more like a http thing. Can you link to the code that was asserting?

Comment (ctiller) [reply src/core/ext/transport/binder/transport/binder_transport.cc:188]: I believe so.

**Comment by TaWeiTu** [comment]
Rebased on master instead so that we can proceed to review & merge this first.

@sifmelcara Can you take a look? Thanks!

**Comment by TaWeiTu** [comment]
Re-triggering the CI since Android build failed.

**Comment by TaWeiTu** [comment]
The failed CIs seem unrelated to this PR.

**Pull request merged**